src/json.cr

```diff
@@ -1,5 +1,51 @@
 # The JSON module allows parsing and generating [JSON](http://json.org/) documents.
 #
+# ### General type-safe interface
+#
+# The general type-safe interface for parsing JSON is to invoke `T.from_json` on a
+# target type `T` and pass either a `String` or `IO` as an argument.
+#
+# ```
+# require "json"
+#
+# json_text = %([1, 2, 3])
+# Array(Int32).from_json(json_text) # => [1, 2, 3]
+#
+# json_text = %({"x": 1, "y": 2})
+# Hash(String, Int32).from_json(json_text) # => {"x" => 1, "y" => 2}
+# ```
+#
+# Serializing is achieved by invoking `to_json`, which returns a `String`, or
+# `to_json(io : IO)`, which will stream the JSON to an `IO`.
+#
+# ```
+# require "json"
+#
+# [1, 2, 3].to_json    # => "[1, 2, 3]"
+# {"x" => 1, "y" => 2} # => "{\"x\": 1, \"y\": 2}
```

Comment:
Missing the `.to_json` call in the example 😁 

```suggestion
# {"x" => 1, "y" => 2}.to_json # => "{\"x\": 1, \"y\": 2}
```

Reply:
Thank you! ❤️